### ISSUE TITLE
Long-lived branch for SIMD

### ISSUE THREAD

**Issue by ryanking13** [pyodide/pyodide#3646]
Related: #1932

Safari 16.4 is going to be released soon. It means that now all major browsers + Node.js + Deno will support WASM fixed-width SIMD:

- Chrome 91
- Firefox 89
- Safari 16.4
- Node.js 16.4
- Deno 1.9

So I think now is a good point start experimenting SIMD... but in a separate branch. Maybe until safari >=16.4 is adopted by most users, and until Node.js drops v14 support.

The good point of SIMD is that we can enable it package-by-package. The main module is not affected, so (hopefully) we only need to fix meta.yaml files. Thus I think syncing and resolving conflicts would not be very hard.

WDYT?

**Comment by rth** [comment]
I think ideally it would be to have SIMD that could be enabled on main via some config flag but disabled by default. And then indeed have a mostly empty separate branch for CI/deployment if needed, changing that config option.

Unless you think that it would be a lot of changes that we cannot/shouldn't merge into main? 

The only Python packages I know of that explicitly uses SIMD would be numpy, unless when you enable that the compiler would also do auto-vectorization which would make code faster in general?

**Comment by rth** [comment]
True, that would be nice. But there are cases where might not want to re-build both wheels (e.g. when testing some other package). So maybe having some global flag wouldn't hurt.

**Comment by ryanking13** [comment]
> Unless you think that it would be a lot of changes that we cannot/shouldn't merge into main?

I think there will a quite small changes, so I think having a global flag would work

> We should build a numpy simd wheel, feature detect simd at runtime and automatically choose the simd wheel if it's available and the no-simd wheel if not.

I am not very tempted to have two wheels which will make our package building and loading system more complicated. So I would prefer to keep SIMD behind a flag and tell people to use it manually if they want until most people update their browsers.

**Comment by hoodmane** [comment]
I don't understand the alternative to using two wheels.

**Comment by ryanking13** [comment]
I was thinking that, for instance, we release SIMD wheels to `cdn.jsdelivr.net/pyodide/dev/simd/` and tell people to use that URL if they want to use SIMD wheels.

**Comment by ryanking13** [comment]
It seems like we don't need a separate branch for testing SIMD. Each package can enable SIMD without affecting the main module.
